gips/0081-indexing-payments.md

```diff
@@ -0,0 +1,165 @@
+
+1. the subgraph to be indexed
+1. the data service that will initiate payment collection
+1. a deadline by which the agreement must be accepted
+1. an end date to the agreement
+1. a maximum amount payable for the initial indexing.
+1. a maximum amount payable for the ongoing indexing. The ongoing indexing is the indexing that happens between the last collection and the present.
+1. the maximum amount of time that can elapse between collections.
+1. a price per unit of work done.
+
+This agreement is to be upheld by protocol smart contracts, leveraging horizon primitives. This means that payers will have to escrow funds similarly to what's done for query fees. Between the smart contracts and the escrow, indexers are guaranteed that if they accept an indexing agreement, they'll be compensated as agreed, without the need to trust the agreement's counterpart.
+
+At the same time, agreements are cancellable by either party at any time, with some limitations put in place to allow indexers to collect for all work done up to that point.
+
+Putting all of this together the solution consists of the following:
+
+- Indexers disclose publicly their price per unit of work.
+- Gateways send indexing agreements to indexers they want to work with.
+- Indexers accept (on-chain) agreements that meet their criteria and get to work indexing.
+- On a regular basis:
+  - Indexers collect for their work, by posting on-chain the original agreement, a Proof of Indexing (PoI) and the amount of work done for that period.
+  - Gateways monitor the QoS per open agreement and assess if they need to cancel (on-chain) and/or replace any of them.
+
+This solution aims to:
+
+- compensate indexers for the exact amount of work they do.
+- allow indexers to gracefully exit the agreement if it can't be fully fulfilled.
+- cap the amount payers will spend on indexing. Though they might spend but get no service back if the cap is too low.
+- incentivize sufficient QoS by awarding agreements to indexers that provide it.
+- provide sufficient transparency and flexibility for gateways, so that they can abstract the complexity and provide simpler pricing to subgraph developers
+
+### Indexing Indexer Selection Algorithm (IISA)
+
+As mentioned above, gateways get to choose who they work with. In Horizon, gateways can be run be any protocol participant and they get to decide how to implement the indexer selection. Nevertheless we propose that a good IISA needs to considers QoS, available stake (with delegation), and other parameters to find the optimal balance between QoS, load-balancing, economic security and price.
+
+### Arbitration and disputes
+
+Since collections are tied to "amount of work done" and this is reported by the indexers themselves, there needs to be a mechanism to punish malicious reporting. We propose here to use slashing. For this to be fair we need the "amount of work done" to be verifiable by an arbitrator in the case of a dispute.
```

Comment:
I am not sure it is going to be verifiable. Maybe you want to leave that can of worms closed... I would keep the emphasis on this low. For example downgrade 'we need' to 'should be'.